In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [14]:
data = pd.read_csv('train.csv')

In [15]:
data.shape

(1460, 81)

The Ames data set contains of 81 columns. This is way to many to make sense of. Let's try to identify which of these attributes have the biggest impact on the house `Sales Price`.

# Null values
First let's study what columns have lot's of null values.

In [17]:
df = data.isnull().sum().sort_values(ascending=False)
df[df>0]

PoolQC          1453
MiscFeature     1406
Alley           1369
Fence           1179
FireplaceQu      690
LotFrontage      259
GarageCond        81
GarageType        81
GarageYrBlt       81
GarageFinish      81
GarageQual        81
BsmtExposure      38
BsmtFinType2      38
BsmtFinType1      37
BsmtCond          37
BsmtQual          37
MasVnrArea         8
MasVnrType         8
Electrical         1
dtype: int64

Some features like PoolQC, MiscFeature, Alley and Fence are missing for over 1000 households (1460 in the dataset). The `data_description.txt` file tells us that this means a lock of pool etc, but still we can't make much of an attribute that is the same for most households.

# Data types

In [21]:
data.dtypes.value_counts()

object     43
int64      35
float64     3
dtype: int64

We have 38 numerical values, and 43 string types. However, without looking at the values it is hard to tell if the variables are categorical or quantitative. A numerical could take limited values and represent a categorical, and a string could represent a date etc.

In [62]:
unique = pd.DataFrame(data.nunique().sort_values(ascending=False)).reset_index().rename(columns={0 : 'unique'})
dtypes =  pd.DataFrame(data.dtypes).reset_index().rename(columns={0 : 'dtype'})
entries =  pd.DataFrame(data.notnull().sum()).reset_index().rename(columns={0 : 'entries'})

In [63]:
unique.head(1)

index  unique
0    Id    1460

In [65]:
df = pd.merge(unique, entries, on = 'index')
df = pd.merge(df, dtypes, on = 'index')

In [66]:
df

index  unique  entries    dtype
0              Id    1460     1460    int64
1         LotArea    1073     1460    int64
2       GrLivArea     861     1460    int64
3       BsmtUnfSF     780     1460    int64
4        1stFlrSF     753     1460    int64
5     TotalBsmtSF     721     1460    int64
6       SalePrice     663     1460    int64
7      BsmtFinSF1     637     1460    int64
8      GarageArea     441     1460    int64
9        2ndFlrSF     417     1460    int64
10     MasVnrArea     327     1452  float64
11     WoodDeckSF     274     1460    int64
12    OpenPorchSF     202     1460    int64
13     BsmtFinSF2     144     1460    int64
14  EnclosedPorch     120     1460    int64
15      YearBuilt     112     1460    int64
16    LotFrontage     110     1201  float64
17    GarageYrBlt      97     1379  float64
18    ScreenPorch      76     1460    int64
19   YearRemodAdd      61     1460    int64
20   Neighborhood      25     1460   object
21   LowQualFinSF      24     1460    int64
22        MiscVal      21     1460    int64
23      3SsnPorch      20     1460    int64
24    Exterior2nd      16     1460   object
25     MSSubClass      15     1460    int64
26    Exterior1st      15     1460   object
27         MoSold      12     1460    int64
28   TotRmsAbvGrd      12     1460    int64
29    OverallQual      10     1460    int64
..            ...     ...      ...      ...
51     GarageCond       5     1379   object
52     GarageQual       5     1379   object
53     GarageCars       5     1460    int64
54         YrSold       5     1460    int64
55     Electrical       5     1459   object
56    FireplaceQu       5      770   object
57       BsmtQual       4     1423   object
58   KitchenAbvGr       4     1460    int64
59       LotShape       4     1460   object
60    LandContour       4     1460   object
61     Fireplaces       4     1460    int64
62       BsmtCond       4     1423   object
63    KitchenQual       4     1460   object
64          Fence       4      281   object
65   BsmtFullBath       4     1460    int64
66    MiscFeature       4       54   object
67      ExterQual       4     1460   object
68   BsmtExposure       4     1422   object
69     MasVnrType       4     1452   object
70       FullBath       4     1460    int64
71         PoolQC       3        7   object
72     PavedDrive       3     1460   object
73       HalfBath       3     1460    int64
74   GarageFinish       3     1379   object
75      LandSlope       3     1460   object
76   BsmtHalfBath       3     1460    int64
77          Alley       2       91   object
78         Street       2     1460   object
79      Utilities       2     1460   object
80     CentralAir       2     1460   object

[81 rows x 4 columns]